jOOQ/src/main/java/org/jooq/tools/jdbc/DefaultStatement.java

```diff
@@ -266,7 +266,13 @@ public boolean getMoreResults(int current) throws SQLException {
 
     @Override
     public ResultSet getGeneratedKeys() throws SQLException {
-        return new DefaultResultSet(getDelegateStatement().getGeneratedKeys(), this);
+        ResultSet rs = getDelegateStatement().getGeneratedKeys();
+        // Some JDBC drivers seem to illegally return null
+        // from getGeneratedKeys() sometimes
+        if (rs != null) {
+            return new DefaultResultSet(rs, this);
+        }
+        return null;
```

Comment:
I wouldn't want to return `null` here, because as we have seen this is a violation of the JDBC API. I would then rather return an empty `ResultSet` (possibly using jOOQ's own `MockResultSet`).

Reply:
Could you elaborate on how to use `MockResultSet` to create a empty `ResultSet`?

Reply:
Thinking more about this, I don't really like the idea of incorporating JDBC driver specific hacks into `DefaultStatement`, as it only implements the delegate pattern.

If the `Settings#withReturnIdentityOnUpdatableRecord()` workaround works for you for now, I think we should leave it at that and then see if we also need to address #8996.